cylc/flow/batch_sys_manager.py

```diff
@@ -133,6 +134,24 @@
 from cylc.flow.parsec.OrderedDict import OrderedDict
 
 
+# Minimal environment to divorce all local jobs from the scheduler environment,
+# to be consistent with remote jobs (not just for background jobs; local batch
+# systems can pass the environment to jobs too). Also stops local jobs from
+# bypassing the central `cylc` wrapper by inheriting the scheduler's virtual
+# environment. (The wrapper gets CYLC_VERSION etc. from the job script, not
+# from the submission environment). (Note this does not affect the higher-level
+# `cylc jobs-submit` command, which does see the scheduler environment and
+# does bypass the "cylc" wrapper).
+# Note also:
+# - $HOME is needed by job.sh.
+# - $HOME may also be needed by /etc/profile.d/ scripts (by observation)
+# TODO - does this need to be a global config setting?
+CLEAN_ENV = {
+    'PATH': '/bin:/usr/bin:/usr/local/bin',
+    'HOME': os.environ['HOME']
+}
```

Comment:
Note I have hardwired a clean minimal `$PATH`.  Is this acceptable, or do we need to:
- extend it to other common bin locations
- or find some universal way of dynamically determining an appropriate minimal `$PATH`
- and/or allow it to be configured in `global.cylc`

See my code comments above - I found on my Centos 7 box that job execution broke if I didn't pass `$HOME` through because `/etc/profile.d/kde.sh` (!) was using `$HOME` as a compulsory argument to a command.  As it happens we need `$HOME` in `job.sh` anyway, but this does suggest the possibility that `CLEAN_ENV` might need to configurable to cover bad login scripts.

Reply:
(What about `sudo pip install cylc-flow` into the system Python ... maybe that should not require a cylc wrapper).

Reply:
Because of non-python dependencies `pip` installing may be too limited and the requirement of sudo access is a problem too.

Could we configure the problem away? I think we used to have a `cylc executable` config, not sure if we kept it.

Reply:
Sorry I wasn't suggesting we require that, just wondering: should that be a supported mode of use (without the web UI) that does not require a wrapper (or special configuration for that matter).  @kinow is concerned that `pip install cylc-flow` should result in a cylc back end that works out of the box for local jobs, which doesn't seem entirely unreasonable.

Reply:
Same goes for conda install, with the UI, actually.  Should users (for personal use or test, say) be able to `conda install cylc` and then run simple workflows "out of the box" without having to install a customized `cylc` wrapper or creating a global config file?

Reply:
(This PR, which we agree was needed for several reasons, will break that use case by preventing local jobs from inheriting the scheduler environment).

Reply:
(Resolving this, I've gone for configurable option so we can have it both ways).